packages/cosmic-proto/src/helpers.ts

```diff
@@ -103,7 +103,7 @@ export const typeUrlToGrpcPath = (typeUrl: Any['typeUrl']) => {
 type RequestQueryOpts = Partial<Omit<RequestQuery, 'path' | 'data'>>;
 
 export const toRequestQueryJson = (
-  msg: QueryBalanceRequestProtoMsg,
+  msg: { value: Uint8Array; typeUrl: string },
```

Comment:
The type described here is `Any` https://github.com/Agoric/agoric-sdk/blob/4c51569c9814fba63865c6de46e4d67497e4be0a/packages/cosmic-proto/src/codegen/google/protobuf/any.ts#L117-L120
```suggestion
  msg: Any,
```

Please maintain use of aliases as documentation of what these types are, that they aren't ad-hoc.

Though I'd encourage keeping `QueryBalanceRequestProtoMsg` since that's the only query we yet support. We can add a union for other messages. I think you only had to make this Any because `buildQueryPacketString` didn't constrain its param, and I think it would be better to instead use `QueryBalanceRequestProtoMsg` in both places as a form of documentation. It doesn't affect future compatibility because it doesn't affect runtime.

If as we add more request types the union of them becomes unwieldy we could define a message type using `RequestTypeUrl`

It you don't use `QueryBalanceRequestProtoMsg` remember to remove its import.

Reply:
The `Any` suggestion is great, thank you. I intended to leverage existing types as much as possible but missed this one. I'm less convinced we need to include `QueryBalanceRequestProtoMsg`, but I've included it in the union anyway.

In somewhat related code I chose to type out `Encoder`. If there's something we can instead import from telescope, that'd be great. We needed something that accepted generic params so we can supply any `*Msg` and get type inference for the second argument (`Partial<T>`) of `buildMsgResponseString` and `buildQueryResponseString`.